onnxruntime/core/providers/dml/DmlExecutionProvider/src/GraphDescBuilder.cpp

```diff
@@ -143,15 +143,40 @@ namespace Dml::GraphDescBuilder
         }
     }
 
-    GraphDesc BuildGraphDesc(
+    uint32_t SetAndGetMainDmlGraphNodeIndex(
+        const uint32_t operatorDmlGraphNodeIndex,
+        const std::string& nodeNamePrefix,
+        AbstractOperatorDesc& operatorDesc,
+        /*in_out*/std::unordered_map<uint32_t, uint32_t>& operatorDmlGraphNodeIndexToMainDmlGraphNodeIndexMap,
+        /*in_out*/std::vector<DmlSerializedGraphNode>& dmlGraphNodes)
+    {
+        auto iter = operatorDmlGraphNodeIndexToMainDmlGraphNodeIndexMap.find(operatorDmlGraphNodeIndex);
+        if (iter != operatorDmlGraphNodeIndexToMainDmlGraphNodeIndexMap.end())
+        {
+            return iter->second;
+        }
+        operatorDmlGraphNodeIndexToMainDmlGraphNodeIndexMap[operatorDmlGraphNodeIndex] = static_cast<uint32_t>(dmlGraphNodes.size());
+        dmlGraphNodes.push_back({operatorDesc, nodeNamePrefix + std::to_string(operatorDmlGraphNodeIndex)});
+        return operatorDmlGraphNodeIndexToMainDmlGraphNodeIndexMap[operatorDmlGraphNodeIndex];
+    }
+
+    /*
+    * Terminology:
+    *   SubGraph: partitioned ONNX graph from the original (main) ONNX graph
+    *   DmlGraph: a graph in DML currency converted from subGraph.
+    *   operatorDmlGraph: a graph in DML currency for a given node or operator
+    * DmlGraph aka mainDmlGraph to distinguish b/w operatorDmlGraph and DmlGraph.
+    */
+    GraphDesc BuildDmlGraphDesc(
         const uint8_t* isConstGpuGraphInput,
         const size_t isConstGpuGraphInputCount,
         const std::unordered_map<std::string, std::pair<const ONNX_NAMESPACE::TensorProto*, bool>>& isInitializerTransferable,
         const onnxruntime::Graph& graph,
         const onnxruntime::IndexedSubGraph& indexedSubGraph,
```

Comment:
>indexedSubGraph [](http://example.com/codeflow?start=44&length=15)

Note subgraph is a single word ([https://www.merriam-webster.com/dictionary/subgraph](https://www.merriam-webster.com/dictionary/subgraph)), like you correctly have it for `constantEdgeIdxToSubgraphInputArgIdxMap`.

Though, it appears this is an existing mistake by ORT with `onnxruntime::IndexedSubGraph` :/ which should have been ``onnxruntime::IndexedSubgraph``. So, I can go either way.